core/ledger/kvledger/txmgmt/validation/batch_preparer.go

```diff
@@ -19,20 +19,95 @@ import (
 	"github.com/hyperledger/fabric/core/ledger/kvledger/txmgmt/rwsetutil"
 	"github.com/hyperledger/fabric/core/ledger/kvledger/txmgmt/statedb"
 	"github.com/hyperledger/fabric/core/ledger/kvledger/txmgmt/txmgr"
-	"github.com/hyperledger/fabric/core/ledger/kvledger/txmgmt/validator"
-	"github.com/hyperledger/fabric/core/ledger/kvledger/txmgmt/validator/internal"
 	"github.com/hyperledger/fabric/core/ledger/util"
 	"github.com/hyperledger/fabric/internal/pkg/txflags"
 	"github.com/hyperledger/fabric/protoutil"
 )
 
+// CommitBatchPreparer performs validation and prepares the final batch that is to be committed to the statedb
+type CommitBatchPreparer struct {
```

Comment:
BlockCommitPreparer may also be overreaching as a block commit has many other things associated with it. `Batch` may not be correct as well... but we have been using it all over for committing the writes to statedb. May be, `StateCommitsPreparer` is a better term?

Reply:
We are trying to exclude invalid transaction and include only valid transaction's write-set. If we can express that, it would be nice. I am unable to find suitable names.

Reply:
`validation.ValidUpdatesRetriever` seems okay to me. This also requires a new API called `validation.ValidUpdatesRetriever.ValidateAndRetrieve()`. The word`valid` occurs thrice which I don't like. Just putting out my thoughts.

Reply:
`validation.ValidUpdatesRetriever.SerializableUpdates()` might be okay.